subprojects/core/src/main/java/org/gradle/execution/TaskPathProjectEvaluator.java

```diff
@@ -50,4 +58,30 @@ public void configureHierarchy(ProjectInternal project) {
             configure((ProjectInternal) sub);
         }
     }
+
+    @Override
+    public void configureHierarchyInParallel(ProjectInternal project) {
+        try {
+            buildOperationExecutor.runAllWithAccessToProjectState(queue -> {
+                for (Project p : project.getAllprojects()) {
+                    queue.add(new RunnableBuildOperation() {
+                        @Override
+                        public void run(BuildOperationContext context) throws Exception {
+                            configure((ProjectInternal) p);
+                        }
+
+                        @Override
+                        public BuildOperationDescriptor.Builder description() {
+                            return BuildOperationDescriptor.displayName("Configure project " + p.getName());
+                        }
+                    });
+                }
+            });
+        } catch (MultipleBuildOperationFailures e) {
```

Comment:
❔ Why do we have to unwrap here?

Reply:
To avoid introducing another exception layer when it's not necessary (and would force us to change some test expectations).

Reply:
It looks like something the build operation executor should do, rather than all the callers being expected to unwrap.

Reply:
I don't like the idea of breaking/changing the contract of `BuildOperationExecutor`. Perhaps this could be changed in the consumer side (whichever piece of code is ultimately dealing with the exception) so the unnecessary layers are removed before reporting.

Incidentally, we're facing the same question for Parallel CC.

Reply:
I don't think we need to do this in the current PR. It would be good to address this in a follow-up. Especially, if there are more cases similar to this.